File: src/cpp_srvcli/src/add_two_ints_server.cpp
#include <memory>

#include "example_interfaces/srv/add_two_ints.hpp"
#include "rclcpp/rclcpp.hpp"

void add(
  const std::shared_ptr<example_interfaces::srv::AddTwoInts::Request> request,
  std::shared_ptr<example_interfaces::srv::AddTwoInts::Response> response)
{
  response->sum = request->a + request->b;

  RCLCPP_INFO(
    rclcpp::get_logger("cpp_add_two_ints_server"),
    "Request: %ld + %ld = %ld",
    request->a, request->b, response->sum);
}

int main(int argc, char ** argv)
{
  rclcpp::init(argc, argv);

  auto node = rclcpp::Node::make_shared("cpp_add_two_ints_server");

  auto service = node->create_service<example_interfaces::srv::AddTwoInts>(
    "cpp_add_two_ints", &add);

  RCLCPP_INFO(node->get_logger(), "Ready to add two ints.");

  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: src/py_srvcli/py_srvcli/add_two_ints_server.py
import rclpy
from example_interfaces.srv import AddTwoInts
from rclpy.node import Node


class AddTwoIntsServer(Node):

    def __init__(self):
        super().__init__('py_add_two_ints_server')
        self.service = self.create_service(
            AddTwoInts,
            'py_add_two_ints',
            self.add_callback,
        )
        self.get_logger().info('Ready to add two ints.')

    def add_callback(self, request, response):
        response.sum = request.a + request.b
        self.get_logger().info(
            f'Request: {request.a} + {request.b} = {response.sum}'
        )
        return response


def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsServer()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()